In [1]:
import pandas as pd
from pandas import json_normalize
import numpy as np
import json as js
import ast as ast
import re

In [2]:
games = pd.read_json('output_steam_games.json',lines=True)

In [3]:
list_rev = []
archivo1 = r'australian_user_reviews.json'
with open(archivo1, encoding='utf-8') as file:
    for line in file.readlines():
        list_rev.append(ast.literal_eval(line))

reviews = pd.DataFrame(list_rev)
reviews = reviews.explode('reviews')

In [4]:
list_items = []
archivo2 = r'australian_users_items.json'
with open(archivo2, encoding='utf-8') as file2:
    for line2 in file2.readlines():
        list_items.append(ast.literal_eval(line2))

items = pd.DataFrame(list_items)
items = items.explode('items')

In [21]:
games.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 120445 entries, 0 to 120444
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   publisher     24083 non-null  object 
 1   genres        28852 non-null  object 
 2   app_name      32133 non-null  object 
 3   title         30085 non-null  object 
 4   url           32135 non-null  object 
 5   release_date  30068 non-null  object 
 6   tags          31972 non-null  object 
 7   reviews_url   32133 non-null  object 
 8   specs         31465 non-null  object 
 9   price         30758 non-null  object 
 10  early_access  32135 non-null  float64
 11  id            32133 non-null  float64
 12  developer     28836 non-null  object 
dtypes: float64(2), object(11)
memory usage: 11.9+ MB


In [22]:
games.tail(20)

publisher                                             genres  \
120425  Spiderweb Software                             [Strategy, RPG, Indie]   
120426    Epic Games, Inc.                                           [Action]   
120427               Valve                                           [Action]   
120428                None                                               None   
120429                None                                               None   
120430                None                                               None   
120431                None                                               None   
120432           Fred Ells                                            [Indie]   
120433             VersoVR  [Casual, Indie, Simulation, Sports, Early Access]   
120434        Phil Fortier                                 [Adventure, Indie]   
120435  Retro Army Limited                         [Action, Adventure, Indie]   
120436  OrtiGames/OrtiSoft                                    [Casual, Indie]   
120437              INGAME                                       [Indie, RPG]   
120438            Riviysky                                    [Casual, Indie]   
120439     Bidoniera Games                 [Action, Adventure, Casual, Indie]   
120440     Ghost_RUS Games              [Casual, Indie, Simulation, Strategy]   
120441              Sacada                          [Casual, Indie, Strategy]   
120442        Laush Studio                        [Indie, Racing, Simulation]   
120443            SIXNAILS                                    [Casual, Indie]   
120444                None                                               None   

                                               app_name  \
120425                                      Geneforge 2   
120426  Unreal Tournament 2004: Editor's Choice Edition   
120427                   Counter-Strike: Condition Zero   
120428        Robotpencil Presents: Exercise: Brushwork   
120429       Robotpencil Presents: Creative Composition   
120430                                 The Gamble House   
120431         Kalen Chock Presents: 2017 Free Tutorial   
120432                          Agent X: Equation Rider   
120433                                     Cricket Club   
120434               Snail Trek - Chapter 3: Lettuce Be   
120435                             The spy who shot me™   
120436                                   Raining blocks   
120437                                          Bravium   
120438                                            BAE 2   
120439                                     Kebab it Up!   
120440                                   Colony On Mars   
120441                         LOGistICAL: South Africa   
120442                                    Russian Roads   
120443                              EXIT 2 - Directions   
120444                                      Maze Run VR   

                                                  title  \
120425                                      Geneforge 2   
120426  Unreal Tournament 2004: Editor's Choice Edition   
120427                   Counter-Strike: Condition Zero   
120428        Robotpencil Presents: Exercise: Brushwork   
120429       Robotpencil Presents: Creative Composition   
120430                                 The Gamble House   
120431         Kalen Chock Presents: 2017 Free Tutorial   
120432                          Agent X: Equation Rider   
120433                                     Cricket Club   
120434               Snail Trek - Chapter 3: Lettuce Be   
120435                             The spy who shot me™   
120436                                   Raining blocks   
120437                                          Bravium   
120438                                            BAE 2   
120439                                     Kebab it Up!   
120440                                   Colony On Mars   
120441                         LOGistICAL: South Africa   
120442       

In [5]:
reviews

user_id                                           user_url  \
0      76561197970982479  http://steamcommunity.com/profiles/76561197970...   
0      76561197970982479  http://steamcommunity.com/profiles/76561197970...   
0      76561197970982479  http://steamcommunity.com/profiles/76561197970...   
1                js41637               http://steamcommunity.com/id/js41637   
1                js41637               http://steamcommunity.com/id/js41637   
...                  ...                                                ...   
25797  76561198312638244  http://steamcommunity.com/profiles/76561198312...   
25797  76561198312638244  http://steamcommunity.com/profiles/76561198312...   
25798        LydiaMorley           http://steamcommunity.com/id/LydiaMorley   
25798        LydiaMorley           http://steamcommunity.com/id/LydiaMorley   
25798        LydiaMorley           http://steamcommunity.com/id/LydiaMorley   

                                                 reviews  
0      {'funny': '', 'posted': 'Posted November 5, 20...  
0      {'funny': '', 'posted': 'Posted July 15, 2011....  
0      {'funny': '', 'posted': 'Posted April 21, 2011...  
1      {'funny': '', 'posted': 'Posted June 24, 2014....  
1      {'funny': '', 'posted': 'Posted September 8, 2...  
...                                                  ...  
25797  {'funny': '', 'posted': 'Posted July 10.', 'la...  
25797  {'funny': '', 'posted': 'Posted July 8.', 'las...  
25798  {'funny': '1 person found this review funny', ...  
25798  {'funny': '', 'posted': 'Posted July 20.', 'la...  
25798  {'funny': '', 'posted': 'Posted July 2.', 'las...  

[59333 rows x 3 columns]

In [6]:
items

user_id  items_count           steam_id  \
0      76561197970982479          277  76561197970982479   
0      76561197970982479          277  76561197970982479   
0      76561197970982479          277  76561197970982479   
0      76561197970982479          277  76561197970982479   
0      76561197970982479          277  76561197970982479   
...                  ...          ...                ...   
88308  76561198329548331            7  76561198329548331   
88308  76561198329548331            7  76561198329548331   
88308  76561198329548331            7  76561198329548331   
88308  76561198329548331            7  76561198329548331   
88309  edward_tremethick            0  76561198331598578   

                                                user_url  \
0      http://steamcommunity.com/profiles/76561197970...   
0      http://steamcommunity.com/profiles/76561197970...   
0      http://steamcommunity.com/profiles/76561197970...   
0      http://steamcommunity.com/profiles/76561197970...   
0      http://steamcommunity.com/profiles/76561197970...   
...                                                  ...   
88308  http://steamcommunity.com/profiles/76561198329...   
88308  http://steamcommunity.com/profiles/76561198329...   
88308  http://steamcommunity.com/profiles/76561198329...   
88308  http://steamcommunity.com/profiles/76561198329...   
88309     http://steamcommunity.com/id/edward_tremethick   

                                                   items  
0      {'item_id': '10', 'item_name': 'Counter-Strike...  
0      {'item_id': '20', 'item_name': 'Team Fortress ...  
0      {'item_id': '30', 'item_name': 'Day of Defeat'...  
0      {'item_id': '40', 'item_name': 'Deathmatch Cla...  
0      {'item_id': '50', 'item_name': 'Half-Life: Opp...  
...                                                  ...  
88308  {'item_id': '373330', 'item_name': 'All Is Dus...  
88308  {'item_id': '388490', 'item_name': 'One Way To...  
88308  {'item_id': '521570', 'item_name': 'You Have 1...  
88308  {'item_id': '519140', 'item_name': 'Minds Eyes...  
88309                                                NaN  

[5170015 rows x 5 columns]

In [40]:
games.isnull().sum()

publisher       8052
genres          3283
app_name           2
title           2050
release_date    2067
tags             163
specs            670
price           1377
early_access       0
id                 2
developer       3299
dtype: int64

In [9]:
items.isnull().sum()

user_id            0
items_count        0
steam_id           0
user_url           0
items          16806
dtype: int64

In [10]:
reviews.isnull().sum()

user_id      0
user_url     0
reviews     28
dtype: int64

In [41]:
items.drop(columns=['user_url'],inplace=True)

In [43]:
reviews.drop(columns=['user_url'],inplace=True)

120444 registros de games antes de borrar duplicados. Son 88309 registros duplicados y vacios. 3135 registros

In [29]:
games.drop(games[games.duplicated(subset=['publisher', 'app_name', 'title', 'url', 'release_date', 'reviews_url', 'price', 'early_access', 'id',
       'developer'],keep=False)].index,inplace=True)

In [34]:
games.drop(columns=['url','reviews_url'],inplace=True)

In [39]:
games['early_access'].value_counts()

early_access
0.0    30188
1.0     1947
Name: count, dtype: int64